pages/dashboard/index.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import _ from 'lodash';
+import getConfig from 'next/config';
+import styled from 'styled-components';
+import { useTranslation } from 'react-i18next';
+
+import { useAuth } from '../../lib/Auth';
+import { useMatrix } from '../../lib/Matrix';
+import ServiceInvitations from './ServiceInvitations';
+import { ServiceTable } from '../../components/UI/ServiceTable';
+import DisplayInvitations from './DisplayInvitations';
+
+const sortRooms = function(room) {
+    return [
+        room.notificationCount === 0,
+        room.name,
+    ];
+};
+
+const TableSection = styled.section`
+  overflow-x: auto;
+
+  tbody tr:hover {
+    background-color: var(--color-background-alpha);
+  }
+`;
+export default function Dashboard() {
+    const auth = useAuth();
+    const matrix = useMatrix(auth.getAuthenticationProvider('matrix'));
+    const MatrixAuthProvider = auth.getAuthenticationProvider('matrix');
+    const { t } = useTranslation('dashboard');
+
+    const invites = matrix.invites;
+    const matrixClient = auth.getAuthenticationProvider('matrix').getMatrixClient();
+    const serviceSpaces = matrix.serviceSpaces;
+    const chatInvitations = _.sortBy([...matrix.invites.values()], sortRooms).filter(invite => !invite.meta);
+
+    // const authProviders = getConfig().publicRuntimeConfig.authProviders; // get the authProvider object from the config
+    // const providersWithTemplates = useMemo(() => {
+    //     return _.pickBy(authProviders, provider => _.has(provider, 'templates')); // only return authProviders which have a templates key
+    // }, [authProviders]);
+
+    // functions which interact with matrix server
+    const declineMatrixInvite = async (e, roomId) => {
+        e.preventDefault();
+        console.log('rejecting ' + roomId);
+        await matrix.leaveRoom(roomId);
+    };
+
+    const acceptMatrixInvite = async (e, roomId, service, name) => {
+        e.preventDefault();
+        await matrixClient.joinRoom(roomId).catch(() => {
+            return;
+        });
+        await MatrixAuthProvider.addSpaceChild(serviceSpaces[service], roomId);
```

Comment:
Sooo... when we accept an invite for e.g. a pad, we add this pad to our "private" Etherpad applications folder? Is that correct? Is this line also being executed for "normal" chats? What does this line do in that case?